dist/js/materialize.js

```diff
@@ -220,8 +217,15 @@ jQuery.extend( jQuery.easing,
-    if (typeof define === 'function' && define.amd) {
-        define(['jquery', 'hammerjs'], factory);
+    if (typeof define === 'function' && define.amd == false) {
+        // hammerjs will already be publicly available
+        define(['jquery'], function($, Hammer) {
```

Comment:
I ran into that require.js error, too, so I edited `dist/materialize.js` and added only line 1 and 220-228 you added.
The reason for that is that I did not want to use a version shipped with materialize but a self-managed using bower.
After installing it and adding it to my require.js configuration everything works fine. However, I had to change line 222 to what I wrote in my comment.

So, even though Hammer might be available publicly I would suggest relying on the AMD dependency management to be more stable in custom environments. However, my idea might be totally wrong since I had not time to check how Hammer is integrated internally.